### PR TITLE
Add Envoy Proxy Prometheus Dashboard

### PR DESCRIPTION
**Pull request by algchoo**
* added dashboard json for envoy proxy prometheus metrics 
* included screenshots 
* included readme

### CODE REVIEW

**Review by EvanSimpson — CHANGES_REQUESTED** [review]
@algchoo It looks like my comments on the Apache Airflow and RabbitMQ dashboard PRs are also applicable here and elsewhere. Can you update your PRs to make sure they follow the contribution guidelines?

**Comment by algchoo** [comment]
> @algchoo It looks like my comments on the Apache Airflow and RabbitMQ dashboard PRs are also applicable here and elsewhere. Can you update your PRs to make sure they follow the contribution guidelines?

@EvanSimpson Thank you for the feedback, I've gone through and made changes to each PR submitted. Please let me know if there is anything else that needs to be changed and I hope you have a great day!

File: dashboards/envoy-proxy/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+sample_dashboards:
+  -
+    category: Envoy
+    id: envoy-proxy
```
Comment (EvanSimpson) [inline dashboards/envoy-proxy/metadata.yaml:4]:
```suggestion
    id: envoy-proxy-prometheus
```

File: dashboards/envoy-proxy/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    display_name: Envoy Proxy
+    description: |-
+      This dashboard has charts displaying Live Proxies, Average Uptime (minutes), Allocated Memory, Heap Size, Unhealty Proxies, Total Active Connections, Total Requests Rate, Allocated Memory, Upstream Network Traffic, CDS Update Attemps, CDS Update Success, and CDS Update Failure
+    related_integrations:
+      - id: envoy
+        platform: GCE
```
Comment (EvanSimpson) [inline dashboards/envoy-proxy/metadata.yaml:10]:
```suggestion
        platform: GKE
```

**Review by EvanSimpson — CHANGES_REQUESTED** [review]

**Comment by EvanSimpson** [comment]
@algchoo Thanks for making the changes! I'll be going through and leaving any more suggestions, as needed.

**Review by EvanSimpson — APPROVED** [review]

**Pull request merged**